Subject: popt: move gettext checks into the right location

AM_GNU_GETTEXT should be run before we test if NLS is being used or not,
so we move the macro to an earlier location. This uncovers some problems
with recent autotools versions which are also being fixed.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 configure.ac |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

Index: configure.ac
===================================================================
--- configure.ac.orig
+++ configure.ac
@@ -1,9 +1,9 @@
 AC_INIT(popt.h)
 AC_CANONICAL_SYSTEM
 AC_PREREQ(2.12)
-AC_CONFIG_HEADERS
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE(popt, 1.10.7)
-AM_CONFIG_HEADER(config.h)
+AC_GNU_SOURCE
 
 # Library code modified:                              REVISION++
 # Interfaces changed/added/removed:   CURRENT++       REVISION=0
@@ -99,7 +99,10 @@ AC_ARG_ENABLE([ld-version-script],
               [ : ] )
 AM_CONDITIONAL(HAVE_LD_VERSION_SCRIPT, test "$have_ld_version_script" = "yes")
 
-if test ! -f ../rpm.c
+AM_GNU_GETTEXT_VERSION([0.16.1])
+AM_GNU_GETTEXT
+
+if test ! -f ../rpm.c -a "$USE_NLS" != "no"
 then
   AC_MSG_CHECKING(for GNU xgettext)
   xgettext --version 2>&1 | grep 'GNU gettext' >/dev/null 2>&1 || AC_MSG_ERROR([
@@ -114,8 +117,6 @@ AC_CHECK_FUNC(setreuid, [], [
 AC_CHECK_FUNCS(getuid geteuid mtrace __secure_getenv setregid strerror)
 
 
-AM_GNU_GETTEXT
-
 POPT_SOURCE_PATH="`pwd`"
 AC_DEFINE_UNQUOTED(POPT_SOURCE_PATH, "$POPT_SOURCE_PATH",
 	[Full path to popt top_srcdir.])
